cpp/conversion.h

```diff
@@ -0,0 +1,706 @@
+#pragma once
```

Comment:
For the sake of rrtmgp integration in other code sources, this file could prob have a more unique name. Something like `rrtmgp_kokkos_integration_utils.hpp` or simply `rrtmgp_kokkos_conversion.h`.

Reply:
Also, minor, but since this is a C++ file, you should probably use `.hpp`. The `.h` file may give the false feeling that it's ok to include it in a C source code..

Reply:
Agreed on both. The pattern in this codebase is for all headers to be `.h` unfortunately, so I just continued with that.